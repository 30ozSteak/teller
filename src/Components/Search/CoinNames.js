import DataCleaner from '../../Utils/Cleaners'

export const coinNames = async () => {
  const cleaner = new DataCleaner()
  console.log(await cleaner.getCoinNames())
  return await cleaner.getCoinNames()
}