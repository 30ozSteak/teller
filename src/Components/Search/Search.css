.search-button,
.search-button-active {
  background: var(--MainGradient);
  border-top: 2px solid transparent;
  bottom: 25px;
  border-radius: 15px;
  box-shadow: 5px 5px 5px rgba(0, 0, 0, 0.256);
  cursor: pointer;
  height: 56px;
  position: fixed;
  right: 35px;
  transform: rotate(45deg);
  width: 56px;
  z-index: 50;
}

.search-button:hover {
  animation: woom 2s linear infinite;
}

.search-button img {
  transition: 0.2s;
}

.search-button:hover img {
  transform: scale(1.1) rotate(-45deg);
}

.search-button-active {
  transform: rotate(0deg);
  transition: 0.9s width;
  width: 56px;
  height: 56px;
  background: none;
  background-color: #303859;
  border-bottom: 2px solid #db5db2;
  box-shadow: 5px 5px 5px rgba(0, 0, 0, 0.434);
  animation: none;
  transition: 0.5s;
  animation: makeBigger 1s forwards;
}

.search-button-active:hover {
  transform: scale(1.05);
}

.search-field input {
  padding: 0.5rem 1rem;
  background: black;
}

.search-bar,
.search-bar-active input {
  border: 0px;
  position: relative;
  bottom: 1.5rem;
  width: 10rem;
  font-size: 1rem;
  letter-spacing: 2px;
  background-color: transparent;
  color: white;
  font-weight: 400;
  text-transform: uppercase;
  transition: 0.5s;
}

.search-button {
  animation: makeSmaller 0.2s forwards;
}

.search-button img {
  transform: rotate(-45deg);
}

.search-button img,
.search-button-active img {
  height: 25px;
  margin: 1rem;
}

.search-button-active img {
  transition: 0.2s;
}

.search-button input {
  opacity: 0;
}

.search-button-active input {
  animation: appear 1.5s forwards;
}

/* ! Animations ! */

@keyframes ❤️ {
  0% {
    box-shadow: 0 0 0 0 rgba(255, 217, 0, 0);
  }
  50% {
    box-shadow: 0 0 0 1px #9862da;
  }
  100% {
    box-shadow: 0 0 0 10px rgba(255, 217, 0, 0);
  }
}

@keyframes appear {
  0% {
    opacity: 0;
  }
  75% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@keyframes makeBigger {
  0% {
    width: 56px;
  }
  80% {
    width: 56px;
  }
  100% {
    width: 20rem;
  }
}

@keyframes woom {
  0% {
    filter: hue-rotate(0);
  }
  50% {
    filter: hue-rotate(20deg);
  }
  100% {
    filter: hue-rotate(0deg);
  }
}
