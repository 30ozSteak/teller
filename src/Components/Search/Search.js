import React, { Component } from "react";
import "./Search.css";
import SearchGlass from "../../Assets/search.svg";
import Cancel from "../../Assets/cancel.svg";

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      active: false,
      hasText: false,
      search: ''
    };
  }

  toggleSearchBtn = () => {
    this.setState({
      active: !this.state.active
    });
  };

  handleChange = (e) => {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }



  render() {
    const { active, search } = this.state;
    return (
      <div className="bigger-box">
        <div className={active ? "search-button-active" : "search-button"}>
          <img
            className="search-image"
            src={active ? Cancel : SearchGlass}
            onClick={this.toggleSearchBtn}
          />
          <input 
            className="search-bar" 
            type="text" 
            placeholder="Search"
            name="search"
            value={search}
            onChange={this.handleChange} />
        </div>
      </div>
    );
  }
}

export default Search;
