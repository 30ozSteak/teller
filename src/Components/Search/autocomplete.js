import { Node } from './Node';
import { coinNames } from './CoinNames'

class Trie {
  constructor() {
    this.rootNode = new Node();
    this.currNode = null;
    this.prevNode = null;
    this.totalWords = 0;
    this.wordList = [];
  }

  insert(word) {
    this.totalWords++;
    let splitString = word.toLowerCase().split('');
    let currNode = this.rootNode;

    splitString.forEach((string, i) => {
      if (currNode.children[string]) {
        currNode = currNode.children[string];
      } else {
        currNode.children[string] = new Node(string);
        currNode = currNode.children[string];
      }

      if (i === splitString.length - 1) {
        currNode.ending = true;
        currNode.finalWord = word;
      }
    });
  }

  count() {
    return this.totalWords;
  }

  suggest(prefix) {
    let splitPrefix = prefix.toLowerCase().split('');
    let currNode = this.traverseTrie(splitPrefix);
    this.findLast(currNode.children);
    // console.log(this.wordList)
    return this.wordList;
  }

  populate(dictionary) {
    dictionary.forEach((word) => {
      this.insert(word.toLowerCase());
      console.log(this.totalWords)
    });
  }

  findLast(childrenObj) {
    let array = Object.keys(childrenObj);
    array.forEach((letter) => {
      let newNode = childrenObj[letter];
      if (newNode.ending === true) {
        this.wordList.push(newNode.finalWord);
      }
      this.findLast(newNode.children);
    });
  }

  traverseTrie(array) {
    let currNode = this.rootNode;
    array.forEach((letter) => {
      if (currNode.children[letter]) {
        currNode = currNode.children[letter];
      }
    });
    return currNode;
  }
}

export default Trie

