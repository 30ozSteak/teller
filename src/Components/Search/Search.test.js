import React from 'react'
import { shallow } from 'enzyme'
import Search from './Search'

describe('Search Component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Search />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should start with the correct default state', () => {
    expect(wrapper.state().active).toEqual(false)
    expect(wrapper.state().hasText).toEqual(false)
    expect(wrapper.state().search).toEqual('')
  })

  describe('toggleSearchBtn function', () => {
    it('should toggle the active property in state', () => {
      const searchBtn = wrapper.find('.search-image')
      searchBtn.simulate('click')
      expect(wrapper.state().active).toEqual(true)
    })
  })

  describe('handleChange function', () => {
    it('should set state with the correct value', () => {
      const searchInput = wrapper.find('.search-bar')
      searchInput.simulate('change', {
        target: { name: 'search', value: 'bitcoin'}
      })
      expect(wrapper.state().search).toEqual('bitcoin')
    })
  })
})