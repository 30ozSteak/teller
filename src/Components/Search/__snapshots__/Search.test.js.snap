// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="search-button"
      >
        <img
          className="search-image"
          onClick={[Function]}
          src="search.svg"
        />
        <input
          className="search-bar"
          name="search"
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
      </div>,
      "className": "bigger-box",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="search-image"
            onClick={[Function]}
            src="search.svg"
          />,
          <input
            className="search-bar"
            name="search"
            onChange={[Function]}
            placeholder="Search"
            type="text"
            value=""
          />,
        ],
        "className": "search-button",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-image",
            "onClick": [Function],
            "src": "search.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-bar",
            "name": "search",
            "onChange": [Function],
            "placeholder": "Search",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="search-button"
        >
          <img
            className="search-image"
            onClick={[Function]}
            src="search.svg"
          />
          <input
            className="search-bar"
            name="search"
            onChange={[Function]}
            placeholder="Search"
            type="text"
            value=""
          />
        </div>,
        "className": "bigger-box",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              className="search-image"
              onClick={[Function]}
              src="search.svg"
            />,
            <input
              className="search-bar"
              name="search"
              onChange={[Function]}
              placeholder="Search"
              type="text"
              value=""
            />,
          ],
          "className": "search-button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "search-image",
              "onClick": [Function],
              "src": "search.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "search-bar",
              "name": "search",
              "onChange": [Function],
              "placeholder": "Search",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
